clash-ghc/src-ghc/Clash/GHC/LoadModules.hs

```diff
@@ -855,8 +869,14 @@ findPrimitiveGuardAnnotations
   -> m [(Text.Text, (PrimitiveGuard ()))]
 findPrimitiveGuardAnnotations bndrs = do
   anns0 <- findNamedAnnotations bndrs
-  let anns1 = errOnDuplicateAnnotations "PrimitiveGuard" bndrs anns0
+  let anns1 = combineAnnotationsWith combinePrimGuards "PrimitiveGuard" bndrs anns0
   pure (map (first (qualifiedNameString' . Var.varName)) anns1)
+ where
+  combinePrimGuards a b = case (a,b) of
+    (HasBlackBox x _, HasBlackBox y _) -> Right (HasBlackBox (x++y) ())
```

Comment:
Does this concatenate the warnings? In that case I think we should do `unlines [x, y]` instead.

Reply:
No, there is a little more structure than that, `x,y :: [PrimitiveWarning]` and 
```haskell
data PrimitiveWarning = WarnNonSynthesizable String | WarnAlways String
```

[This `go`](https://github.com/clash-lang/clash-compiler/blob/3a2d3209788aa440356f0c39ae088ef720da6562/clash-lib/src/Clash/Netlist/BlackBox.hs#L382) function goes over that resulting `[PrimitiveWarning]` and calls [`warn`](https://github.com/clash-lang/clash-compiler/blob/3a2d3209788aa440356f0c39ae088ef720da6562/clash-lib/src/Clash/Netlist/BlackBox.hs#L110) on them when applicable.
And `warn` then does a `hPutStLn` for each one.

Reply:
ahhhhh